private-templates-service/adaptivecards-templating-service/src/storageproviders/MongoDBProvider.ts

```diff
@@ -32,8 +32,10 @@ export class MongoDBProvider implements StorageProvider {
         })
       };
     }
-
-    return templateQuery;
+    if(templateQuery.name && templateQuery.tags){
+      return {$or: [{name:templateQuery.name}, {tags:templateQuery.tags}]}
+    }
+    return templateQuery
```

Comment:
Or condition is correct but partial name/tag search is not going to return correct results. If you look a little bit up in this function, you'll see how partial search is done for name and tags. Try to put that in. I might be mistaken though and your approach will work just fine.
The **$all** operator for tags above is used to find templates that have all the specified tags inside (they might have more than that). https://docs.mongodb.com/manual/reference/operator/query/all/